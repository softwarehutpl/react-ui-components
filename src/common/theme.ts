const theme: { [key: string]: { [key: string]: { [key: string]: string } } } = {
  colors: {
    primary: {
      base: '#000000',
      light: '#ffffff',
      dark: '#a8b3bd',
    },
    secondary: {
      base: '#3690e3',
      light: '#75b0e6',
      dark: '#034d91',
    },
    error: {
      base: '#f44336',
      light: '#f56c62',
      dark: '#a1140a',
    },
    warning: {
      base: '#faa72d',
      light: '#ffc36b',
      dark: '#995b00',
    },
    info: {
      base: '#7979d4',
      light: '#9797d1',
      dark: '#4c4cc7',
    },
    success: {
      base: '#4caf50',
      light: '#88c28a',
      dark: '#124f14',
    },
    disabled: {
      base: '#d9d9d9',
      light: '#f0eded',
      dark: '#8c8989',
    },
  },
  // TODO: change it in dedicated task for that (change ease-out for example and add more options)
  transitions: {
    all: {
      fast: 'all 0.5s ease-out',
      mid: 'all 2s ease-out',
      slow: 'all 4s ease-out',
    },
    background: {
      fast: 'background-color 0.5s ease-out',
      mid: 'background-color 2s ease-out',
      slow: 'background-color 4s ease-out',
    },
    fontColor: {
      fast: 'color 0.5s ease-out',
      mid: 'color 2s ease-out',
      slow: 'color 4s ease-out',
    }
  }
};

export default theme;
