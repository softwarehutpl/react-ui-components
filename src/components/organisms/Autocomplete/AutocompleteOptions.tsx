import React from 'react';
import styled, { css } from 'styled-components';

export interface OptionItemProps {
  isActive?: boolean;
  isHover?: boolean;
  color?: string;
  hoverOption?: number;
};

export interface OptionWrapperProps {
  width?: number;
  height?: number;
  padding?: number;
  margin?: number;
}

export interface AutoCompleteOptionsProps extends OptionWrapperProps, OptionItemProps {
  options: string[];
  onClick?: (e: React.MouseEvent<HTMLLIElement>) => void;
  onMouseEnter?: any;
  onMouseLeave?:(e: React.MouseEvent<HTMLLIElement>) => void;
  activeOption?: number;
  hoverOption?: number;
};

export const OptionListWrapper = styled.ul<OptionWrapperProps>`
  width: ${({ width }) => (width ? `${width}px` : '100%' )};
  height: ${({ height }) => (height ? `${height}px` : 'auto' )};
  overflow: auto;
  box-shadow: 0px 0px 10px rgba(0,0,0,0.12);
  border-radius: 15px;
  padding: ${({ padding }) => `${padding}px` };
  margin:${({ margin }) => `${margin}px` };
`;

export const OptionItem = styled.li<OptionItemProps>`
  list-style-type: none;
  padding: 15px;
  background-color: ${({ color, theme, isActive }) => ((isActive && color) && theme.colors[color].dark)};
  color: ${({ theme, color }) => (color && theme.colors[color].base)};
  cursor: pointer;

  ${({ isHover, color }) => css`
    background-color: ${({ theme }) => ((color && isHover ) && theme.colors[color].dark)};
  `}
`;

const AutocompleteOptions: React.FunctionComponent<AutoCompleteOptionsProps> = ({
  options,
  onClick,
  onMouseEnter,
  onMouseLeave,
  activeOption,
  color,
  width,
  height,
  padding,
  margin,
  hoverOption,
}) => (
  <OptionListWrapper
    height={height}
    width={width}
    padding={padding}
    margin={margin}
  >
    {options.map((option: string, index: number) => (
      <OptionItem
        key={option}
        onClick={onClick}
        onMouseEnter={(e) => onMouseEnter(e, index)}
        onMouseLeave={onMouseLeave}
        isActive={activeOption === index}
        color={color}
        isHover={hoverOption === index}
      >
        {option}
      </OptionItem>
    ))}
  </OptionListWrapper>
);

AutocompleteOptions.defaultProps = {
  padding: 0,
  margin: 0,
};

export default AutocompleteOptions;
