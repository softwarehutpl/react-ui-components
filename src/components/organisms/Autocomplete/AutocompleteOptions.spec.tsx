import React from 'react';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';
import AutocompleteOptions, { 
  OptionListWrapper,
  OptionItem 
} from './AutocompleteOptions';
import theme from '../../../common/theme';

const options = [
  'Papaya',
  'Apple',
  'Pear',
  'Peach',
  'Pomegranate',
  'Pineapple',
  'Banana',
];

describe('AutocompleteOptions', () => {
  it('should render corectly', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <AutocompleteOptions options={options} />
      </ThemeProvider>).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should have proper style when isActive props is true', () => {
    const component = create(
      <OptionItem
        color="primary"
        theme={theme}
        isActive
      />
    ).toJSON();
    expect(component).toHaveStyleRule('background-color', '#f6f6f6');
    expect(component).toHaveStyleRule('color', '#000000');
  });

  it('should have proper width and height if provided', () => {
    const component = create(
      <OptionListWrapper
        color="primary"
        theme={theme}
        width={100}
        height={100}
      />
    ).toJSON();
    expect(component).toHaveStyleRule('width', '100px');
    expect(component).toHaveStyleRule('height', '100px');
  });
});
