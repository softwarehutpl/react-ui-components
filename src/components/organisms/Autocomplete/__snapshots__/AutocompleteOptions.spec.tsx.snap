// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutocompleteOptions should render corectly 1`] = `
.c0 {
  width: 100%;
  height: auto;
  overflow: auto;
  box-shadow: 0px 0px 10px rgba(0,0,0,0.12);
  border-radius: 15px;
  padding: 0px;
  margin: 0px;
}

.c1 {
  list-style-type: none;
  padding: 15px;
  cursor: pointer;
}

<ul
  className="c0"
>
  <li
    className="c1"
    onMouseEnter={[Function]}
  >
    Papaya
  </li>
  <li
    className="c1"
    onMouseEnter={[Function]}
  >
    Apple
  </li>
  <li
    className="c1"
    onMouseEnter={[Function]}
  >
    Pear
  </li>
  <li
    className="c1"
    onMouseEnter={[Function]}
  >
    Peach
  </li>
  <li
    className="c1"
    onMouseEnter={[Function]}
  >
    Pomegranate
  </li>
  <li
    className="c1"
    onMouseEnter={[Function]}
  >
    Pineapple
  </li>
  <li
    className="c1"
    onMouseEnter={[Function]}
  >
    Banana
  </li>
</ul>
`;
